Bowling.java

```diff
@@ -2,25 +2,91 @@
 
 public class Bowling {
 
-  private int totalPoints;
+  private Scoreboard scoreboard;
 
   public Bowling() {
-    totalPoints = 0; 
+    scoreboard = new Scoreboard();
+    scoreboard.addFrame();
   }
 
-  public Bowling bowl(int pinsDown) {
-		if (pinsDown < 0) {
-			throw new IllegalArgumentException("Cannot knock down negative pins. Bowl again.");
-		}
-    totalPoints += pinsDown;
+  public Bowling bowl(Integer pinsDown) {
+    if (pinsDown < 0) {
+      throw new IllegalArgumentException("Cannot knock down negative pins. Bowl again.");
+    }
+
+    Frame currentFrame = scoreboard.getCurrentFrame();
```

Comment:
I had a feeling that I was violating encapsulation, but I couldn't see how before. Your suggestion and the article provided more clarity. There is one part I'd like to make more concrete tomorrow when we talk in person, but this is one part I'm holding on to: "...according to the Law of Demeter for Methods, any method of an object should only call methods belonging to: itself, any parameters that were passed in to the method, any objects it created, any composite objects." I've been struggling with knowing when its OK to allow objects to 'talk'.

Questions: 1) Is isFull() poor? If so, why? 2) Is the Scoreboard an inappropriate object? If so, why? 

Thanks for explaining more about the PP and TDD process! I will push the changes I made which remove knowledge of Frame from the Bowling class early tomorrow if not tonight.